### PR TITLE
Performance: Fix Headers base-lining to simple implementation 

### PR DESCRIPTION
**Pull request by kirankumarkolli**
MicroBenchmarking results on a 4core Linux VM. 

CosmosBenchmark:

> ReadStreamV3 resutls
> --database testdb --container testcol --partitionkeypath /pk -n 200000 -w ReadStreamExistsV3 --pl 300

| Measurement |   Mater RPS |  PR RPS | %Change |
|------------ | ----------- |--------:|--------:|
| TOP  10% AVG | 11034 | 9281  | 18.9|
| TOP  20% AVG | 10932 | 8664  | 26.2|
| TOP  30% AVG | 10701 | 8396  | 27.5|
| TOP  40% AVG | 10438 | 8320  | 25.5|
| TOP  50% AVG | 10326 | 8175  | 26.3|
| TOP  60% AVG | 10153 | 8068  | 25.8|
| TOP  70% AVG | 9883  | 8018  | 23.3|
| TOP  80% AVG | 9778  | 7918  | 23.5|
| TOP  90% AVG | 9692  | 7824  | 23.9|
| TOP  95% AVG | 9556  | 7768  | 23|

Microbenchmark:
- Memory allocations are almost down by 50%
- Mean times improved by 50-60%
- P95 improved by 35-40%

|              Method |     Mean |        P95 |       P100 | Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------------------- |---------:|-----------:|-----------:|------:|------:|------:|----------:|
|          CreateItem | 807.0 us | 1,665.3 us | 2,530.6 us |     - |     - |     - |  38.59 KB |
|    DeleteItemExists | 833.5 us | 1,598.1 us | 2,335.6 us |     - |     - |     - |  38.65 KB |
| DeleteItemNotExists | 993.7 us | 1,846.6 us | 2,663.2 us |     - |     - |     - |   55.8 KB |
|            ReadFeed | 954.8 us | 1,676.9 us | 2,549.6 us |     - |     - |     - |  51.94 KB |
|      ReadItemExists | 928.7 us | 1,864.9 us | 2,358.2 us |     - |     - |     - |  43.84 KB |
|   ReadItemNotExists | 044.4 us | 1,904.8 us | 2,476.7 us |     - |     - |     - |   60.8 KB |
|          UpdateItem | 819.7 us | 1,577.1 us | 2,138.2 us |     - |     - |     - |   39.1 KB |
|          UpsertItem | 853.2 us | 1,859.6 us | 2,731.2 us |     - |     - |     - |  38.69 KB |

|              Method |     Mean |        P95 |       P100 | Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------------------- |---------:|-----------:|-----------:|------:|------:|------:|----------:|
|          CreateItem | 382.8 us | 1,058.9 us | 1,743.2 us |     - |     - |     - |  20.16 KB |
|    DeleteItemExists | 375.3 us |   941.3 us | 1,727.7 us |     - |     - |     - |  20.23 KB |
| DeleteItemNotExists | 534.3 us | 1,154.9 us | 1,579.8 us |     - |     - |     - |   37.2 KB |
|            ReadFeed | 581.8 us | 1,191.3 us | 1,569.1 us |     - |     - |     - |  33.09 KB |
|      ReadItemExists | 513.9 us | 1,205.1 us | 1,881.5 us |     - |     - |     - |  25.27 KB |
|   ReadItemNotExists | 679.7 us | 1,608.9 us | 2,369.4 us |     - |     - |     - |  42.23 KB |
|          UpdateItem | 409.9 us | 1,063.5 us | 1,577.2 us |     - |     - |     - |  20.45 KB |
|          UpsertItem | 405.5 us |   912.9 us | 1,545.4 us |     - |     - |     - |   20.3 KB |

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Headers/CosmosMessageHeadersInternal.cs [hunk 1]
```diff
@@ -18,13 +18,10 @@ namespace Microsoft.Azure.Cosmos
     /// </summary>
     internal class CosmosMessageHeadersInternal : INameValueCollection
     {
-        private readonly Lazy<Dictionary<string, string>> headers = new Lazy<Dictionary<string, string>>(CosmosMessageHeadersInternal.CreateDictionary);
+        private readonly Dictionary<string, string> headers = CosmosMessageHeadersInternal.CreateDictionary();
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Headers/CosmosMessageHeadersInternal.cs:21]: Can we add the perf test and numbers into this PR to show the benefit of this change? That way people in the future will know what to run to validate they didn't cause a regression.

File: Microsoft.Azure.Cosmos/Directory.Build.props [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <ClientOfficialVersion>3.9.1</ClientOfficialVersion>
+    <ClientOfficialVersion>3.10.1</ClientOfficialVersion>
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/Directory.Build.props:4]: Is this picking up other changes?

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/Directory.Build.props]: Seems like

File: Microsoft.Azure.Cosmos/src/Headers/CosmosMessageHeadersInternal.cs (hunk 1 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Headers/CosmosMessageHeadersInternal.cs]: PR refreshed with numbers

**Comment by j82w** [comment]
Closing since there has been no update for over 30 days. Please re-open if necessary.

**Pull request closed without merging**